ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -0,0 +1,92 @@
+# limitations under the License.
+
+import os
+import platform
+
+import rosdistro
+
+
+def print_platform_info():
+    """Print out platform information."""
+    # platform info
+    print('System Information')
+    print('system   : ', platform.system())
+    print('Linux distribution   : ', platform.dist())
+    print('Mac OS version       : ', platform.mac_ver())
+    print('release  : ', platform.release())
+    print('processor: ', platform.processor())
+    print('\n')
+
+    # python info
+    print('Python')
+    print('version      : ', platform.python_version())
+    print('compiler     : ', platform.python_compiler())
+    print('build        : ', platform.python_build())
+    print('\n')
+
+
+def print_ros2_reqs():
+    """Print out ROS2 distribution info using `rosdistro`."""
+    distro_name = os.environ.get('ROS_DISTRO').lower()
+    u = rosdistro.get_index_url()
+    i = rosdistro.get_index(u)
+    distro_info = i.distributions.get(distro_name)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+
+    print('ROS Information')
+    print('distribution name    : ', distro_name)
+    print('distribution type    : ', distro_info.get('distribution_type'))
+    print('distribution status  : ', distro_info.get('distribution_status'))
+    print('release platforms    : ', distro_data.get('release_platforms'))
+    print('\n')
+
+
+def check_platform():
+    """Check platform information against ROS2 requirements."""
+    distro_name = os.environ.get('ROS_DISTRO').lower()
+    u = rosdistro.get_index_url()
+    i = rosdistro.get_index(u)
+    distro_info = i.distributions.get(distro_name)
+    distro_data = rosdistro.get_distribution(i, distro_name).get_data()
+
+    # check distro status
+    if distro_info.get('distribution_status') == 'prerelease':
+        print('WARNING: Distribution is not fully supported or tested.\
+            To get more stable features,\
+                Download a stable version at\
+                    https://index.ros.org/doc/ros2/Installation/')
+    elif distro_info.get('distribution_status') == 'end-of-life':
+        print('WARNING: Distribution is no longer supported or deprecated.\
+            To get the latest features,\
+                Download the latest version at\
+                    https://index.ros.org/doc/ros2/Installation/')
+    else:
+        pass
+
+    # check system platform and distro release platforms
+    supported_platforms = distro_data.get('release_platforms')
+    if platform.system() == 'Linux':
+        releases = supported_platforms.get(platform.dist()[0].lower())
+        if not releases or platform.dist()[2].lower() not in releases:
+            print('WARNING: Current system platform is not supported\
+                by this ROS distribution.\
+                    User `ros2 doctor platform -r` to check report for detail.')
+        else:
+            pass
+    else:
+        print('WARNING:\
```

Comment:
I would recommend not printing this. If a check can't be done on a platform, then it should be silently skipped since there's nothing that can be done by the user to fix it.